### PR TITLE
Added support for exit sleep mode

### PR DESCRIPTION
**Pull request by viktor1970**
Sometimes can be useful put the display in sleep mode. Setting command "thup=1", the display is waked up by a touch event.
The touch event is sent to microcontroller with X and Y coordinates, but the original poll() code does not manage that event.

I added the capability of managing that event in a defined page and component ID, so a callback function can be called.

This because microcontroller needs to know (or it should know!) if the display is in sleep mode or not.

There are 3 new #define in NextionTypes.h:

#define NEX_SS    : code supports Screen Saver (using sleep) mode? Comment it for no support
#define NEX_SS_COMP XX  : touching component ID XX will activare event to exit Screen Saver mode (event attached will be called)
#define NEX_SS_PAGE YY    : the page YY in which component is

Note: if no callback defined, the display exits sleep mode but, obviously, nothing is called.

The most common use is to create an entire screen touch area and attach a callback. When display is in sleep mode and user touch the screen, the callback is called so microcontroller "knows" that (for example) can restart to display data.

### CODE REVIEW

**Comment by viktor1970** [comment]
Note: wake() method can be used, but the screen needs to be touched 2 times: the 1st only exits sleep mode (with no callback), the 2nd to activate a callback (so microcontroller knows end of screen saver)

File: NextionTypes.h [hunk 1]
```diff
@@ -3,6 +3,12 @@
 #ifndef __NEONEXTION_NEXTIONTYPES
 #define __NEONEXTION_NEXTIONTYPES
 
+
+#define NEX_SS true		//The code supports Screen Saver (using sleep) mode?
+#define NEX_SS_COMP 1	//Touching that component ID will activare event to exit Screen Saver mode (event attached will be called)
+#define NEX_SS_PAGE 0	//The page in which component is
```
Comment (DanNixon) [inline NextionTypes.h]: This should not be hardcoded, what if I want the component to be on page 7?

File: README.md [hunk 2]
```diff
@@ -13,16 +13,13 @@ Avaliable through the library manager of the Arduio IDE as `NeoNextion`.
 
 ## New features
```
Comment (DanNixon) [inline README.md]: Please remove changes to this file.

File: Nextion.cpp [hunk 3]
```diff
@@ -68,6 +70,19 @@ void Nextion::poll()
         }
       }
     }
+#ifdef NEX_SS
```
Comment (DanNixon) [inline Nextion.cpp]: Why is this conditionally compiled?

File: Nextion.cpp [hunk 4]
```diff
@@ -56,7 +58,7 @@ void Nextion::poll()
         for (i = 1; i < 7; i++)
           buffer[i] = m_serialPort.read();
         buffer[i] = 0x00;
-
+		Serial.println(String(buffer[0])+" "+String(buffer[1])+" "+String(buffer[2])+" "+String(buffer[3])+" "+String(buffer[4])+" "+String(buffer[5])+" "+String(buffer[6]));
```
Comment (DanNixon) [inline Nextion.cpp]: Please remove use of the `String` class.

File: INextionWidget.h [hunk 5]
```diff
@@ -26,13 +26,14 @@ class INextionWidget
...
+  uint8_t m_pageID;      //!< ID of page this widget is on
+  
+  protected:
   bool sendCommand(char *commandStr, bool checkComplete = true);
 
 protected:
   Nextion &m_nextion;    //!< Reference to the Nextion driver
-  uint8_t m_pageID;      //!< ID of page this widget is on
-  uint8_t m_componentID; //!< Component ID of this widget
   const char *m_name;    //!< Name of this widget
```
Comment (DanNixon) [inline INextionWidget.h]: Why is this changed?

File: NextionTypes.h (hunk 1 above)
Comment (viktor1970) [reply NextionTypes.h]: If the component is on page 7 (or any other page)

#define NEX_SS_PAGE 7

But you're right, I'll try to create method for setting at runtime.

File: README.md (hunk 2 above)
Comment (viktor1970) [reply README.md]: Ok

File: Nextion.cpp (hunk 4 above)
Comment (viktor1970) [reply Nextion.cpp]: Ops I left the serial prints (use for debug).

File: Nextion.cpp (hunk 3 above)
Comment (viktor1970) [reply Nextion.cpp]: Because if there' no need for SS wake (commenting #define NEX_SS), the code is not needed.

File: INextionWidget.h (hunk 5 above)
Comment (viktor1970) [reply INextionWidget.h]: Because I need to access to the properties from poll() section (to compare component and page id)

Comment (DanNixon) [reply INextionWidget.h]: There are public getters for that. This is just breaking encapsulation.

File: Nextion.cpp (hunk 3 above)
Comment (DanNixon) [reply Nextion.cpp]: There is no real need to have any conditional compilation for different features.

**Comment by viktor1970** [comment]
Fixed the code. I'll open another PR
